architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,206 @@
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| mTLS                  | Mutual Transport Layer Security (mTLS) allows two parties to authenticate each other during the initial connection of an SSL/TLS handshake.                                                 |
+| UCP                 | Universal Control Plane for Radius                                                                                                                                                   |
+## System Description
+
+Applications RP is a Radius service that acts as resource provider for application and its resources. The resources can be core resources like application or environment or container. They can also be a dapr resource, message queue or datastore. Applications RP lives in `radius-system` namespace in a kubernetes cluster. It is a client of Controller and UCP. It also receives requests from UCP for managing the above mentioned resources. 
+
+### Architecture
+
+Applications RP has four types of resource providers for managing various types of resources in an application. 
+
+`Applications.Core` resource provider manages application, environment, container and gateways. 
+
+`Applications.Dapr` resource provider manages all dapr resources that are deployed as part of application. 
+
+`Applications.Datastore` resource provider manages datastore such as SQL database or Mongo DB that an application might be using. 
+
+`Applications.Messaging` resources provider manages queues such as Rabbit MQ. 
+
+RecipeEngine is a key sub-component of Applications RP, which uses the above resource providers for provisioning infrastructure resources. It takes as input a recipe, written in bicep or terraform. It fetches Bicep recipe from any OCI complaint registry. It also fetches Terraform recipes available as public modules. It then requests UCP to help deploy the recipe. Also, if user creates a custom recipe, it requests Controller to first validate the recipe using recipe-webhook and then create it. 
+
+Users could provision application resources without using recipes too, in which case, Applications RP again works with Controller managing for Kubernetes resources and UCP for managing AWS and Azure resources. For deploying Terraform Recipe, it directly communicates with AWS and Azure.
+
+The Applications RP provisions some resources synchronously, whereas for other resources whose creation can be time consuming, it has workers that enable async operation. To faciliate Async operation, the Applications RP adds the incoming request to an in-memory queue. Workers dequeue requests from the queue and process them. 
```

Comment:
UCP adds the messages for the async operations to the queue. Worker and Queue belong to UCP.